src/common/colorlabels.c

```diff
@@ -135,8 +132,7 @@ typedef enum dt_colorlabels_actions_t
 
 static void _colorlabels_execute(const GList *imgs, const int labels, GList **undo, const gboolean undo_on, const int action)
 {
-  GList *images = (GList *)imgs;
-  while(images)
+  for(GList *images = (GList *)imgs; images; images = g_list_next(images))
```

Comment:
`  for(const GList *images = imgs; images; images = g_list_next(images))
`?

Reply:
Sure, but then we need a cast on the g_list_next call, since it takes a non-const arg.
I've made the change.

Reply:
Are you sure that was needed? `g_list_next` is just a macro that translates into `(GList *)(list))->next)` so I wouldn't expect a cast to be required and indeed Clang at least doesn't ask for one. Does gcc?

Reply:
Remind  me not to post first thing in the morning when I'm not yet fully awake.  Misremembered having  to add the cast to a g_list_first in one of the loops....

Reply:
I suffer from the same syndrome! Coffee is a very unreliable cure (sometimes it makes you _think_ faster, sometimes it makes you _post_ faster).

It was kinda funny that in the other two changes in the same commit you happily had the const in the type _without_ feeling the need for an explicit cast in g_list_next; so...  ;-)